packages/SwingSet/src/liveslots/virtualObjectManager.js

```diff
@@ -478,16 +478,38 @@ export function makeVirtualObjectManager(
     return obj;
   }
 
-  function bindMethods(context, behaviorTemplate) {
-    const obj = {};
-    for (const prop of Reflect.ownKeys(behaviorTemplate)) {
-      const func = behaviorTemplate[prop];
+  function bindMethods(tag, contextMap, behaviorMethods) {
+    const prototype = {};
+    for (const prop of Reflect.ownKeys(behaviorMethods)) {
+      const func = behaviorMethods[prop];
       assert.typeof(func, 'function');
-      const method = (...args) => Reflect.apply(func, null, [context, ...args]);
+
+      // Violating all Jessie rules to create representative that inherit
+      // methods from a shared prototype. The bound method therefore needs
+      // to mention `this`. We define it using concise menthod syntax
+      // so that it will be `this` sensitive but not constructable.
+      //
+      // We normally consider `this` unsafe because of the hazard of a
+      // method of one abstraction being applied to an instance of
+      // another abstraction. To prevent that attack, the bound method
+      // checks that it is in the map in which such its representatives
+      // are registered.
```

Comment:
Garbled. I think this is an improvement.
```suggestion
      // checks that it is in the map in which its representatives
      // are registered.
```

Reply:
Done, thanks.